src/preamble.js

```diff
@@ -643,8 +643,10 @@ Module['FS_createPreloadedFile'] = FS.createPreloadedFile;
 #include "URIUtils.js"
 
 #if ASSERTIONS
-function createExportWrapper(name, fixedasm) {
-  return function() {
+function createExportWrapper(name, mangled, fixedasm) {
+  // We create a wrapper that all calls get routed through, for the lifetime of the program.
+
+  globalThis[mangled] = Module[mangled] = function() {
```

Comment:
Why is `globalThis` needed here? Aside from that, this PR looks very promising to me!

But it would also be good to see why this doesn't help pyodide - seems like it should.